ads/nativery.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {validateData, writeScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function nativery(global, data) {
+  validateData(data, ['wid']);
+  const params = Object.assign({}, data);
+
+  // push the two object into the '_nativery' global
+  window._nativery =
+    window._nativery ||
+    Object.assign(
+      {},
+      {
+        wid: data.wid,
+        referrer: data.referrer || window.context.referrer,
+        url: data.url || window.context.canonicalUrl,
+        viewId: window.context.pageViewId,
+        params,
+      }
+    );
+
+  // must add listener for resize
+  window.addEventListener('amp-widgetCreated', function(e) {
```

Comment:
Not sure if I have seen this event. What is it meant for?

Reply:
Hi @powerivq, It's a custom event to adjust the height based on the content